packages/node/src/indexer/blockDispatcher/worker-block-dispatcher.service.ts

```diff
@@ -112,7 +110,15 @@ export class WorkerBlockDispatcherService
 
     const blockAmount = await this.projectService.getProcessedBlockCount();
     this.setProcessedBlockCount(blockAmount ?? 0);
-    this.runtimeService = runtimeService;
+
+    // copy Runtime service to each worker
+    this.workers.map((w) =>
+      w.syncRuntimeService(
```

Comment:
If this is only called once, why not make it part of the `init` worker function so that its clearer. The way I read this function is that it should be called each time the state of the main thread runtime service changes.